web/packages/teleport/src/types.ts

```diff
@@ -114,6 +114,7 @@ export interface TeleportFeature {
   isLocked?(lockedFeatures: LockedFeatures): boolean;
   lockedNavigationItem?: TeleportFeatureNavigationItem;
   lockedRoute?: TeleportFeatureRoute;
+  hideNavigation?: boolean;
```

Comment:
Could you add JSDoc to `hideNavigation` and `hideFromNavigation` which explains how they differ from each other?

I think the names makes sense once you understand what navigation is. The comments would be helpful mostly to people looking at this code for the first time.